src/reducers/vis-state-merger.js

```diff
@@ -52,43 +51,49 @@ export function mergeFilters(state, filtersToMerge) {
 
   // merge filters
   filtersToMerge.forEach(filter => {
-    // match filter.dataId with current datesets id
-    // uploaded data need to have the same dataId with the filter
-    if (datasets[filter.dataId]) {
-      // datasets is already loaded
-      const validateFilter = validateFilterWithData(
-        datasets[filter.dataId],
-        filter
-      );
+    const datasetIds = Object.keys(datasets).reduce((acc, datasetId) => ([
```

Comment:
There is one scenario when 1 filter applies to multiple dataset ['a', 'b']. however, only 'a' is loaded at the moment. We should handle that scenario. in your current implementation, this filter will be sent to unmerged. because `validateFilterWithData`